IPython/terminal/pt_inputhooks/qt.py

```diff
@@ -1,15 +1,29 @@
 import sys
+import os
 from IPython.external.qt_for_kernel import QtCore, QtGui
 
 # If we create a QApplication, keep a reference to it so that it doesn't get
 # garbage collected.
 _appref = None
-
+_already_warned = False
 
 def inputhook(context):
     global _appref
     app = QtCore.QCoreApplication.instance()
     if not app:
+        if sys.platform == 'linux':
+            if not os.environ.get('DISPLAY') \
+                    and not os.environ.get('WAYLAND_DISPLAY'):
+                import warnings
+                global _already_warned
+                if not _already_warned:
```

Comment:
I don't think we need this check - the warnings module already takes care of only printing a warning the first time it occurs.

Reply:
```python
In [1]: import matplotlib.pyplot

.../ipython/IPython/terminal/pt_inputhooks/qt.py:22: UserWarning: The DISPLAY or WAYLAND_DISPLAY enviroment variable is not set or empty and Qt5 requires this enviroment variable. Deactivate Qt5 code.
  'The DISPLAY or WAYLAND_DISPLAY enviroment variable is '
In [2]: 

.../ipython/IPython/terminal/pt_inputhooks/qt.py:22: UserWarning: The DISPLAY or WAYLAND_DISPLAY enviroment variable is not set or empty and Qt5 requires this enviroment variable. Deactivate Qt5 code.
  'The DISPLAY or WAYLAND_DISPLAY enviroment variable is '
In [2]: 

.../ipython/IPython/terminal/pt_inputhooks/qt.py:22: UserWarning: The DISPLAY or WAYLAND_DISPLAY enviroment variable is not set or empty and Qt5 requires this enviroment variable. Deactivate Qt5 code.
  'The DISPLAY or WAYLAND_DISPLAY enviroment variable is '
In [2]: 
```

I do not know how the warning module works, but at least on my system it does not work properly. When I disable the `_already_warned` I get above output (I pressed enter) with multiple warnings. When I activate it, I only get one warning.